api/pkg/block/service.go

```diff
@@ -74,24 +235,95 @@ func (s *APIService) CreateVolume(request *restful.Request, response *restful.Re
 	response.WriteEntity(res.Volume)
 }
 
-// List all volumes
-func (s *APIService) ListVolumes(request *restful.Request, response *restful.Response) {
-	if !policy.Authorize(request, response, "volume:list") {
+func (s *APIService) UpdateVolume(request *restful.Request, response *restful.Response) {
+	if !policy.Authorize(request, response, "volume:put") {
+		return
+	}
+	log.Info("Received request for updating volume.")
+
+	volume := &model.Volume{}
+
+	err := request.ReadEntity(&volume)
+	if err != nil {
+		log.Errorf("Failed to read request body err: \n", err)
+		response.WriteError(http.StatusInternalServerError, err)
 		return
 	}
-	log.Infof("Received request to list volumes")
 
-	backend := request.QueryParameter(common.REQUEST_PATH_BACKEND_ID)
+	vol := &block.Volume{
+		Description: volume.Description,
+	}
+
+	if volume.Metadata != nil || len(volume.Metadata) != 0 {
```

Comment:
If volume.metadata is nil then why to check its length?

Reply:
metadata can be empty also.
Please check the logic which is to check map for NULLorEmpty
https://www.javaguides.net/2019/04/check-if-map-is-null-or-empty-in-java.html

Reply:
Thanks for the link.
But in case of golang, even checking len(volume.Metadata) != 0 should be fine, as len of nil is 0